KAFKA-15050: format the prompts in the quickstart

joobisb opened the pull request:
This patch formats the prompts in quickstart and in kafka streams quickstart

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

joobisb committed:
KAFKA-15050: format the prompts in the quickstart

tombentley commented on docs/quickstart.html:
```diff
@@ -215,19 +215,17 @@ <h4 class="anchor-heading">
             Edit the <code class="language-bash">config/connect-standalone.properties</code> file, add or change the <code>plugin.path</code> configuration property match the following, and save the file:
```
We don't actually mean it "should match the following" if _the following_ is some bash commands. 

```suggestion
            Edit the <code class="language-bash">config/connect-standalone.properties</code> file, adding or changing the <code>plugin.path</code> configuration property to include <code>libs/connect-file-{{fullDotVersion}}.jar</code> and save the file. Alternatively, in your terminal session you can use the following command:
```

tombentley commented on docs/quickstart.html:
```diff
@@ -215,19 +215,17 @@ <h4 class="anchor-heading">
 
-        <pre class="brush: bash;">
-&gt; echo "plugin.path=libs/connect-file-{{fullDotVersion}}.jar"</pre>
+        <pre class="line-numbers"><code class="language-bash">$ echo "plugin.path=libs/connect-file-{{fullDotVersion}}.jar"</code></pre>
```
```suggestion
        <pre class="line-numbers"><code class="language-bash">$ echo "plugin.path=libs/connect-file-{{fullDotVersion}}.jar" >> config/connect-standalone.properties</code></pre>
```

tombentley commented on docs/streams/quickstart.html:
```diff
@@ -91,8 +91,8 @@ <h4><a id="quickstart_streams_download" href="#quickstart_streams_download">Step
 
-<pre class="line-numbers"><code class="language-bash">&gt; tar -xzf kafka_{{scalaVersion}}-{{fullDotVersion}}.tgz
-&gt; cd kafka_{{scalaVersion}}-{{fullDotVersion}}</code></pre>
+  <pre class="line-numbers"><code class="language-bash">$ tar -xzf kafka_{{scalaVersion}}-{{fullDotVersion}}.tgz
```
Do we need the initial two spaces?

joobisb replied:
right will remove those

tombentley reviewed (commented):
Thanks! I left a few comments.

joobisb committed:
KAFKA-15050: address comments

joobisb commented:
@tombentley have addressed the comments, please have a look

fvaleri reviewed (approved):
Thanks for fixing this.

I generated a local preview and compared with the old revision.

LGTM.

tombentley commented:
@mimaison mentioned in a separate conversation a couple of other points:

1. it looks like we never need to refer to any of the example lines of console input/output using the line numbers, so perhaps we can get rid of the line number for shell command blocks (to it's less visually busy). 
2. the console producer example is also missing the `>` prompts, maybe we can add them as part of this PR too

joobisb committed:
Merge remote-tracking branch 'origin/trunk' into KAFKA-15050

joobisb committed:
KAFKA-15050: addressed comments

joobisb commented:
> @mimaison mentioned in a separate conversation a couple of other points:
> 
> 1. it looks like we never need to refer to any of the example lines of console input/output using the line numbers, so perhaps we can get rid of the line number for shell command blocks (to it's less visually busy).
> 2. the console producer example is also missing the `>` prompts, maybe we can add them as part of this PR too

@tombentley @mimaison I've addressed the comments, please have a look

tombentley commented on docs/quickstart.html:
```diff
@@ -154,9 +154,9 @@ <h4 class="anchor-heading">
-This is my first event
-This is my second event</code></pre>
+        <pre class="language-bash"><code class="language-bash">$ bin/kafka-console-producer.sh --topic quickstart-events --bootstrap-server localhost:9092
+$ This is my first event
+$ This is my second event</code></pre>
```
`kafka-console-producer.sh` has its own prompt using a `>` character and without a space.

```suggestion
>This is my first event
>This is my second event</code></pre>
```

joobisb replied:
will update these

joobisb replied:
resolved

tombentley commented on docs/quickstart.html:
```diff
@@ -32,7 +32,7 @@ <h4 class="anchor-heading">
         </p>
 
-        <pre class="line-numbers"><code class="language-bash">$ tar -xzf kafka_{{scalaVersion}}-{{fullDotVersion}}.tgz
+        <pre class="language-bash"><code class="language-bash">$ tar -xzf kafka_{{scalaVersion}}-{{fullDotVersion}}.tgz
```
The `<code>` already has `class="language-bash"`, so I don't think we need it on the `<pre>` as well.

joobisb replied:
<img width="684" alt="image" src="https://github.com/apache/kafka/assets/13068144/e4285029-5a1c-4002-8ed6-f8a95c97e2a0">

If I remove the `language-bash` class from `<pre>` it will lose the associated properties as shown in the first image above

tombentley replied:
It works fine for me with `<pre><code class="language-bash">`, using both FF and a chrome-based browser.

tombentley reviewed (commented):
Thanks. I found a couple more nits.

joobisb committed:
KAFKA-15050: addressed comments

joobisb commented:
@tombentley could you please have a look, have addressed the comments

tombentley commented on docs/quickstart.html:
```diff
@@ -154,9 +154,9 @@ <h4 class="anchor-heading">
-This is my first event
-This is my second event</code></pre>
+        <pre class="language-bash"><code class="language-bash">$ bin/kafka-console-producer.sh --topic quickstart-events --bootstrap-server localhost:9092
+> This is my first event
+> This is my second event</code></pre>
```
The console producer doesn't actually include a space after the `>`.

tombentley commented on docs/streams/quickstart.html:
```diff
@@ -152,8 +150,8 @@ <h4><a id="quickstart_streams_prepare" href="#quickstart_streams_prepare">Step 3
-<pre class="line-numbers"><code class="language-bash">&gt; bin/kafka-topics.sh --create \
-    --bootstrap-server localhost:9092 \
+<pre class="language-bash"><code class="language-bash">$ bin/kafka-topics.sh --create \
+--bootstrap-server localhost:9092 \
```
We should keep the indentation of this line (as we do for the following continuation lines).

tombentley reviewed (commented):
@joobisb sorry for the long delay on this one. I found a few nits.

joobisb committed:
Merge remote-tracking branch 'origin/trunk' into KAFKA-15050

joobisb committed:
KAFKA-15050: updated comments

joobisb commented:
@tombentley i've addressed the comments, please have a look

tombentley commented:
@joobisb when I said

> It works fine for me with `<pre><code class="language-bash">`, using both FF and a chrome-based browser.

I meant that I still don't think we need the changes which add the `class="language-bash"` to `<pre>`. Can you remove those?

Sorry for not being clear.

joobisb committed:
Merge remote-tracking branch 'origin' into KAFKA-15050

joobisb committed:
remove class from pre tag

joobisb commented:
> @joobisb when I said
> 
> > It works fine for me with `\<pre\><code class="language-bash">`, using both FF and a chrome-based browser.
> 
> I meant that I still don't think we need the changes which add the `class="language-bash"` to `\<pre\>`. Can you remove those?
> 
> Sorry for not being clear.

@tombentley I have removed the `class="language-bash"` from `\<pre\>`. Could you please take a look

joobisb commented:
@tombentley @flavray 

I have addressed all the comments, can we merge this ?
